tensorflow/core/graph/graph_constructor_test.cc

```diff
@@ -3205,6 +3205,56 @@ TEST_F(GraphConstructorTest, ImportGraphDef_ValidateColationConstraints) {
   TF_EXPECT_OK(ImportGraphDef(options, def, &graph_, nullptr));
 }
 
+TEST_F(GraphConstructorTest, ImportGraphDef_ValidateDefaultDevice) {
+  std::string gdef_ascii(
+      R"EOF(
+      node { name: 'test_input' op: 'TestInput' }
+      node { name: 'test_op' op: 'TestMul' input: [ 'test_input:0', 'test_input:1' ] }
+      )EOF");
+
+  GraphDef gdef;
+  CHECK(protobuf::TextFormat::ParseFromString(gdef_ascii, &gdef));
+
+  ImportGraphDefOptions options;
+  std::string dev = "/gpu:13";
+  options.default_device = dev;
+  options.return_nodes = std::vector<std::string>{"test_input", "test_op"};
+
+  ImportGraphDefResults res;
+
+  TF_EXPECT_OK(ImportGraphDef(options, gdef, &graph_, NULL, &res));
+  EXPECT_EQ(res.return_nodes.size(), options.return_nodes.size());
+  for (auto node: res.return_nodes) {
+    EXPECT_EQ(node->requested_device(), dev);
+  }
+}
+
+TEST_F(GraphConstructorTest, ImportGraphDef_ValidateDefaultDeviceWithAlreadyPresentedDevice) {
```

Comment:
Sorry to nitpick, but instead of making a whole new test, can you just add a device to one of the NodeDefs in the above test? (Or add another NodeDef to the above test.) The return_nodes are returned in the order specified, so for example, if you add "some dev" to the first node, you can check that the first returned node has device "some dev" and the second node "/gpu:13".

Reply:
Sure, I've updated test, it now includes both assigned and graph_def-stored devices test